servers/rendering/renderer_scene_cull.cpp

```diff
@@ -3155,7 +3201,7 @@ void RendererSceneCull::_render_scene(const RendererSceneRender::CameraData *p_c
 		cull_data.visible_layers = p_visible_layers;
 		cull_data.render_reflection_probe = render_reflection_probe;
 		cull_data.occlusion_buffer = RendererSceneOcclusionCull::get_singleton()->buffer_get_ptr(p_viewport);
-		cull_data.camera_matrix = &p_camera_data->main_projection;
+		cull_data.camera_matrix = &p_camera_data->shadow_projection;
```

Comment:
Why is it correct for culling to use the plain perspective projection?

If the intent is to use this for planar reflections, wouldn't a frustum camera projection be more appropriate for culling purposes?

Reply:
The oblique camera is only implemented as essentially a 'trick' version of the Perspective camera. This is because it was developed with games that use that camera in mind.

As mentioned in an above comment, the `shadow_projection` as overhead is only in service to this one camera, for every other type it will be a duplicate of the `main_projection`, which is why I call it overhead.

I don't know enough about the rendering pipeline or lighting system to understand why using the Oblique projection breaks the shadows (a problem with my first PR for this feature), but my guess is that for every *_Projection, any attempt to use the Oblique_*_Projection variation will require the shadows and lighting to use the original *_Projection.

This is why I feel that adding the boolean toggle for all cameras would justify the overhead.

Reply:
yeah I get the idea of adding a boolean to enable the overhead. I can't entirely say I understand if it's correct for light/shadow to use a separate projection matrix (might it also add complexity to the fragment shader?)

the comment I am trying to make here is that culling could be based on the frustum. I show an example here of how using the frustum projection for culling can be helpful: https://github.com/godotengine/godot/pull/85529#issuecomment-1915745899
it would be good to double check if occlusion culling works correctly since the origin point in a classical perspective projection might be outside the room.... while using a frustum might make this check more correct.

Reply:
Culling currently assumes the near and far planes are parallel. Passing the non-oblique projection probably avoids some troubles there. Now, the consequence is that objects between the oblique and the non-oblique planes won't be culled on the CPU (they will be on the GPU anyway). This is probably an acceptable trade-off though.